buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,53 @@
+plugins {
+  `java-gradle-plugin`
+  `kotlin-dsl`
+  id("com.diffplug.spotless") version "5.12.4"
+}
+
+spotless {
+  java {
+    googleJavaFormat("1.10.0")
+    licenseHeaderFile(rootProject.file("../gradle/enforcement/spotless.license.java"), "(package|import|public)")
+    target("src/**/*.java")
+  }
+}
+
+gradlePlugin {
+  plugins {
+    create("muzzle-plugin") {
+      id = "muzzle"
+      implementationClass = "io.opentelemetry.instrumentation.gradle.muzzle.MuzzlePlugin"
+    }
+  }
+}
+
+repositories {
+  mavenCentral()
+  gradlePluginPortal()
+  mavenLocal()
+}
+
+tasks.withType<Test>().configureEach {
+  useJUnitPlatform()
+}
+
+dependencies {
+  implementation(gradleApi())
+  implementation(localGroovy())
+
+  implementation("org.eclipse.aether:aether-connector-basic:1.1.0")
+  implementation("org.eclipse.aether:aether-transport-http:1.1.0")
+  implementation("org.apache.maven:maven-aether-provider:3.3.9")
+
+  implementation("com.google.guava:guava:30.1-jre")
+  implementation("gradle.plugin.com.github.jengelman.gradle.plugins:shadow:7.0.0")
+  implementation("org.ow2.asm:asm:7.0-beta")
+  implementation("org.ow2.asm:asm-tree:7.0-beta")
+  implementation("org.apache.httpcomponents:httpclient:4.5.10")
+  // When updating, also update dependencyManagement/dependencyManagement.gradle.kts
```

Comment:
/cc @iNikem I bundled in an update to the byte buddy gradle plugin into this PR (it didn't fix the problem I was seeing but was my first try :P)